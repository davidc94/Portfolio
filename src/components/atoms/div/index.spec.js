import React from 'react';
import renderer from 'react-test-renderer';
import StyledDiv from './index';

describe('StyledDiv component', () => {
  it('render default StyledDiv as expected', () => {
    const render = renderer.create(
      <StyledDiv>I have nothing to offer</StyledDiv>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('render StyledDiv with a different border color', () => {
    const render = renderer.create(
      <StyledDiv borderColor="header">nice border color</StyledDiv>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('render StyledDiv with a different border width', () => {
    const render = renderer.create(
      <StyledDiv borderWidth="large">damn this border extra thicc</StyledDiv>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('render StyledDiv with a different border width and border color', () => {
    const render = renderer.create(
      <StyledDiv borderWidth="medium" borderColor="border">wooooaaaahhhh</StyledDiv>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});