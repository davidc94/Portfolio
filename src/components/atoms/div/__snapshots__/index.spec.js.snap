// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledDiv component render StyledDiv with a different background color 1`] = `
<div
  className="sc-bdVaJa kTBsKK"
>
  nice background color
</div>
`;

exports[`StyledDiv component render StyledDiv with a different border color 1`] = `
<div
  className="sc-bdVaJa fIeexc"
>
  nice border color
</div>
`;

exports[`StyledDiv component render StyledDiv with a different border radius 1`] = `
<div
  className="sc-bdVaJa lblMvb"
>
  ooh, curvy
</div>
`;

exports[`StyledDiv component render StyledDiv with a different border width 1`] = `
<div
  className="sc-bdVaJa eAUtwn"
>
  damn this border extra thicc
</div>
`;

exports[`StyledDiv component render StyledDiv with a different border width and border color 1`] = `
<div
  className="sc-bdVaJa hXJjeI"
>
  wooooaaaahhhh
</div>
`;

exports[`StyledDiv component render StyledDiv with a different width 1`] = `
<div
  className="sc-bdVaJa bNveZT"
  width="full"
>
  w i d e
</div>
`;

exports[`StyledDiv component render default StyledDiv as expected 1`] = `
<div
  className="sc-bdVaJa frronE"
>
  I have nothing to offer
</div>
`;
